### PR TITLE
CLI using cobra

### PR DESCRIPTION
**Pull request by areed**
Subcommands are implemented as methods on runners type, which holds
dependencies.
Integration tests with ginkgo run against an API.

### CODE REVIEW

File: cli/cmd/channel_rm.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// channelRmCmd represents the channelRm command
+var channelRmCmd = &cobra.Command{
+	Use:   "rm <id>",
+	Short: "Remove (archive) a channel",
+	Long:  `replicated channel rm 4d3d240ea1ec4dab0be3b2105ff4b4ed`,
```
Comment (emosbaugh) [inline cli/cmd/channel_rm.go]: lets remove long descriptions that echo command

File: cli/cmd/channel_inspect.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	Use:   "inspect",
+	Short: "Show full details for a channel",
+	Long:  `replicated channel inspect be52315888f23408e2e4dc9242d4cc2c`,
+}
+
+func init() {
+	channelCmd.AddCommand(channelInspectCmd)
+}
+
+func (r *runners) channelInspect(cmd *cobra.Command, args []string) error {
```
Comment (emosbaugh) [inline cli/cmd/channel_inspect.go]: im concerned about how parseable this output is. a lot of ppl will use cli for automation

File: cli/cmd/channel_create.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	channelCreateCmd.Flags().StringVar(&channelCreateDescription, "description", "", "A longer description of this channel")
+}
+
+func (r *runners) channelCreate(cmd *cobra.Command, args []string) error {
+	allChannels, err := r.api.CreateChannel(r.appID, channelCreateName, channelCreateDescription)
+	if err != nil {
+		return err
+	}
+
+	return print.Channels(r.w, allChannels)
```
Comment (emosbaugh) [inline cli/cmd/channel_create.go]: would be awesome if printers supported formats eg tabs, json, yaml

File: cli/cmd/channel_inspect.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/replicatedhq/replicated/cli/print"
+)
+
+// channelInspectCmd represents the channelInspect command
+var channelInspectCmd = &cobra.Command{
+	Use:   "inspect",
+	Short: "Show full details for a channel",
+	Long:  `replicated channel inspect be52315888f23408e2e4dc9242d4cc2c`,
```
Comment (emosbaugh) [inline cli/cmd/channel_inspect.go]: lets remove long descriptions that echo command

**Comment by areed** [comment]
I fixed the short/long/use help text. Still todo:
* docs  from godoc and generated like on-prem cli
* print yaml or json
* accept appID in addition to slug
* split channel inspect into multiple commands with more parseable output

**Comment by emosbaugh** [comment]
@areed wanna create issues in github?

**Comment by areed** [comment]
added

**Pull request merged**